cmake/compile_tests/amd_apu.cc

```diff
@@ -0,0 +1,38 @@
+//@HEADER
+// ************************************************************************
+//
+//                        Kokkos v. 4.0
+//       Copyright (2022) National Technology & Engineering
+//               Solutions of Sandia, LLC (NTESS).
+//
+// Under the terms of Contract DE-NA0003525 with NTESS,
+// the U.S. Government retains certain rights in this software.
+//
+// Part of Kokkos, under the Apache License v2.0 with LLVM Exceptions.
+// See https://kokkos.org/LICENSE for license information.
+// SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+//
+//@HEADER
+
+#include <iostream>
+#include <hip/hip_runtime_api.h>
+
+int main() {
+  hipDeviceProp_t hipProp;
+  hipError_t error = hipGetDeviceProperties(&hipProp, 0);
+
+  if (error != hipSuccess) {
+    std::cout << hipGetErrorString(error) << '\n';
+    return error;
+  }
+
+  if (hipProp.integrated == 1) {
+    // We detected an APU
+    std::cout << "ON";
```

Comment:
On the MI300A I have access to, `hipProp.integrated == 0` when using rocm 6.1. 
But it works great with rocm 6.2.

Reply:
It looks like `hipProp.integrated` was added in [ROCm 6.2](https://rocm.docs.amd.com/en/docs-6.2.0/about/release-notes.html#id3)... It seems that before that version there is no way to know if you are using an APU.